packages/ember-core/src/components/form/search.gts

```diff
@@ -0,0 +1,424 @@
+    evt.stopPropagation();
+
+    if (this.activeIndex < this.options.length - 1) {
+      this.activeIndex++;
+      this.scrollActiveOptionIntoView();
+    }
+  }
+
+  @action
+  enterKeyHandler(evt?: KeyboardEvent) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+
+    if (this.activeIndex == -1) {
+      return;
+    }
+
+    const option = this.internalOptions[this.activeIndex];
+
+    if (option != undefined) {
+      this.selectOption(option, this.activeIndex);
+    }
+
+    this.onBlur();
+  }
+
+  @action
+  exitKeyHandler(evt?: KeyboardEvent) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+
+    this.onBlur();
+  }
+
+  @action
+  onFocus(evt: FocusEvent) {
+    evt.preventDefault();
+    evt.stopPropagation();
+
+    this.isFocused = true;
+  }
+
+  @action
+  onBlur() {
+    this.isFocused = false;
+    this.searchInputElement?.blur();
+  }
+
+  @action
+  onSearch(evt: Event) {
+    this.searchString = (evt.target as HTMLInputElement).value;
+    this.value = '';
+    this.activeIndex = -1;
+
+    if (!this.canPerformSearch) {
+      return;
+    }
+
+    this.onQuery.perform(this.searchString);
+  }
+
+  @action
+  clear() {
+    this.searchString = '';
+    this.value = '';
+    this.activeIndex = -1;
+
+    this.onBlur();
+  }
+
+  @action
+  onSearchBarInsert(element: HTMLElement) {
+    this.searchInputElement = element;
+  }
+
+  @action
+  onMenuInsert(element: HTMLElement) {
+    this.menuElement = element;
+  }
+
+  <template>
+    <div class={{this.classList}} {{onClickOutside this.onBlur}} ...attributes>
+      <div class="input-group">
+        {{#unless this.hideSearchIcon}}
+          <span class="input-group-text">
+            {{#if this.loading}}
+              <span class="spinner-border spinner-border-sm" />
+            {{else}}
+              <i class="bi bi-search" />
+            {{/if}}
+          </span>
+        {{/unless}}
+        <input
+          aria-describedby={{@describedBy}}
+          id={{@id}}
+          class={{this.inputClassList}}
+          disabled={{@disabled}}
+          readonly={{@readonly}}
+          placeholder={{this.placeholder}}
+          type="text"
+          value={{this.displayValue}}
+          {{on "input" this.onSearch}}
+          {{on "focus" this.onFocus}}
+          {{onKey "ArrowUp" this.moveUp onlyWhenFocused=true}}
+          {{onKey "ArrowDown" this.moveDown onlyWhenFocused=true}}
+          {{onKey "Enter" this.enterKeyHandler onlyWhenFocused=true}}
+          {{onKey "NumpadEnter" this.enterKeyHandler onlyWhenFocused=true}}
+          {{onKey "Tab" this.exitKeyHandler onlyWhenFocused=true}}
+          {{onKey "Escape" this.exitKeyHandler onlyWhenFocused=true}}
+          {{onInsert this.onSearchBarInsert}}
+        />
+        {{#if this.clearable}}
+          <button
+            class="input-group-text"
+            type="button"
+            {{on "click" this.clear}}
+          >
+            <i class="bi bi-x-lg" />
+          </button>
```

Comment:
I would recommend using an NRG `Button` instead - you also don't need the `input-group-text`, buttons are automatically formatted correctly.
```suggestion
          <Button
            aria-label="Clear"
            class="btn-outline-secondary"
            @onClick={{this.clear}}
          >
            <i class="bi bi-x-lg" />
          </Button>
```

Lastly, you'll want to add an `aria-label` attribute for accessibility, and you'll need to translate it with the `t` helper.